os/include/tinyara/sched.h

```diff
@@ -86,18 +86,24 @@
 /********************************************************************************
  * Pre-processor Definitions
  ********************************************************************************/
-#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(CONFIG_ARM_MPU)
+#if defined(CONFIG_ARM_MPU) && (defined(CONFIG_APP_BINARY_SEPARATION) || defined (CONFIG_MPU_STACKGUARD))
+#define REGS_PER_REGION 3
 enum {
 	REG_RNR,
 	REG_RBAR,
 	REG_RASR
 };
 
+#if defined(CONFIG_MPU_STACKGUARD)
```

Comment:
We already checked for CONFIG_MPU_STACKGUARD at line 89. So, this check is not required.